crates/uv-resolver/src/marker.rs

```diff
@@ -92,7 +92,13 @@ fn string_is_disjoint(this: &MarkerExpression, other: &MarkerExpression) -> bool
 ///
 /// This is useful in cases where creating conjunctions or disjunctions might occur in a non-deterministic
 /// order. This routine will attempt to erase the distinction created by such a construction.
-pub(crate) fn normalize(tree: MarkerTree) -> Option<MarkerTree> {
+pub(crate) fn normalize(mut tree: MarkerTree) -> Option<MarkerTree> {
+    filter_all(&mut tree);
+    normalize_all(tree)
```

Comment:
Are there redundant expressions that we'd only notice after normalizing (e.g., flattening)? In other words, is `filter_all(normalize_all(...))` idempotent? Or is there some risk that we don't normalize everything by way of this ordering?

Reply:
`filter_all` was written to handle non-flattened trees, and also leaves the tree in a denormalized (not flattened) state.

Thinking about it more though, I suppose it is possible that there are duplicates that only show up after combining version ranges, so maybe we have to normalize before *and* after? I can try to create a test case where this happens.

Reply:
Fixed and added a test.